mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,300 @@
+        artifacts[artifact_file_name] = artifact
+
+    def _log_pandas_df_artifact(
+        self, artifacts, temp_dir, pandas_df, run_id, artifact_name, dataset_name
+    ):
+        client = mlflow.tracking.MlflowClient()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.csv"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pandas_df.to_csv(artifact_file_local_path, index=False)
+        client.log_artifact(run_id, artifact_file_local_path)
+        artifact = CsvEvaluationArtifact(
+            uri=get_artifact_uri(run_id, artifact_file_name),
+            content=pandas_df,
+        )
+        artifact.load(artifact_file_local_path)
+        artifacts[artifact_file_name] = artifact
+
+    def _log_model_explainality(self, artifacts, temp_dir, model, X, dataset_name, run_id, sample_ratio):
+        model_loader_module = model.metadata.flavors['python_function']["loader_module"]
+        if model_loader_module == 'mlflow.sklearn':
+            raw_model = model._model_impl
+        elif model_loader_module == 'mlflow.lightgbm':
+            raw_model = model.lgb_model
+        elif model_loader_module == 'mlflow.xgboost':
+            raw_model = model.xgb_model
+        else:
+            raw_model = None
+
+        if len(X) < _MIN_SAMPLE_ROWS_FOR_SHAP:
+            sample_rows = len(X)
+        else:
+            sample_rows = max(int(len(X) * sample_ratio), _MIN_SAMPLE_ROWS_FOR_SHAP)
+
+        sampled_X = shap.sample(X, sample_rows)
+        if raw_model:
+            maskers = shap.maskers.Independent(sampled_X)
+            if shap.explainers.Linear.supports_model_with_masker(raw_model, maskers):
+                explainer = shap.explainers.Linear(raw_model, maskers)
+            elif shap.explainers.Tree.supports_model_with_masker(raw_model, maskers):
+                explainer = shap.explainers.Tree(raw_model, maskers)
+            elif shap.explainers.Additive.supports_model_with_masker(raw_model, maskers):
+                explainer = shap.explainers.Additive(raw_model, maskers)
+            else:
+                explainer = shap.explainers.Sampling(model.predict, X)
+        else:
+            explainer = shap.explainers.Sampling(model.predict, X)
+
+        if isinstance(explainer, shap.explainers.Sampling):
+            shap_values = explainer(X, sample_rows)
+        else:
+            shap_values = explainer(sampled_X)
+
+        def plot_summary():
+            shap.plots.beeswarm(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_summary, run_id, "shap_summary", dataset_name
+        )
+
+        def plot_feature_importance():
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts,
+            temp_dir,
+            plot_feature_importance,
+            run_id,
+            "shap_feature_importance",
+            dataset_name,
+        )
+
```

Comment:
We should probably add the `shap.summary_plot()` visualization here. It is a JS plot so would have to be saved as html, though.

Reply:
added.
I don't find easy way to export matplot as html, but the `shap.summary_plot()` could also exported as 'png' image and works fine. The result image is similar with the `shap.plots.beeswarm` output.

Reply:
Since `shap.summary_plot` and `shap.plots.beeswarm` log similar result, shall we log only one of them ?

Reply:
logged both as discussed offline